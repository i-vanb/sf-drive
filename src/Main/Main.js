import React, {useState} from "react";
import "../styles/Main.css";
import main_car_mob from "../img/main-car-mob.svg";
import main_2 from "../img/main_2.svg";
import main_3 from "../img/main_3.svg";
import main_4 from "../img/main_4.svg";
import birka from "../img/birka.svg";
import dollar from "../img/dollar.svg";
import percent from "../img/percent.svg";
import money from "../img/money.svg";
import toy_car from "../img/toy_car.svg";
import Footer from "../components/Footer";
import {NavLink} from "react-router-dom";


function MainUnAuth() {
    return (
        <>
            <section className="main">
                <div className="content-container">
                    <img className="main_img_mob" src={main_car_mob} alt="main picture" />
                    <h1>Каршеринг в любой</h1>
                    <h1>точке России</h1>
                    <br/>
                    <p className="main__text">
                        Будьте всегда за рулём во время путешествий
                    </p>
                    <p className="main__text">
                        и командировок.
                    </p>
                    <div className="register__main__header">
                        <NavLink to="/register/1">
                            Зарегистрироваться
                        </NavLink>
                    </div>
                </div>
            </section>
            <section>
                <div className="content-container row">
                    <div>
                        <img src={main_2} alt="main picture" />
                    </div>
                    <div className="desc">
                        <h2>Аренда напрямую
                            от владельцев</h2>
                        <p>
                            Вы получите автомобиль от его собственника,
                            а мы проверим юридическую чистоту и техническую исправность.
                        </p>
                    </div>
                </div>
            </section>
            <section>
                <div className="content-container row">
                    <div className="desc">
                        <h2>Автомобили
                            на любой вкус
                        </h2>
                        <p>
                            Вы всегда можете подобрать автомобиль любого класса от бюджетных моделей до
                            премиум-класса и спорткаров.
                        </p>
                    </div>
                    <div>
                        <img src={main_3} alt="main picture" />
                    </div>
                </div>
            </section>
            <section>
                <div className="content-container row">
                    <div>
                        <img src={main_4} alt="main picture" />
                    </div>
                    <div className="desc">
                        <h2>Гарантия честной аренды
                        </h2>
                        <p>
                            Общение и оплата происходит через наш сервис, что предотвращает вас от обмана.
                        </p>
                    </div>
                </div>
            </section>
            <section className="bg-gray">
                <div className="content-container">
                    <h2>Как арендовать автомобиль</h2>
                    <div className="circle_li__wrap">
                        <div className="circle_li__item_container">
                            <span className="circle_li__item_img">1</span>
                            <span className="circle_li__item_text">Выберите автомобиль</span>
                        </div>
                        <div className="circle_li__item_container">
                            <span className="circle_li__item_img">2</span>
                            <span className="circle_li__item_text">Забронируйте дату и время</span>
                        </div>
                        <div className="circle_li__item_container">
                            <span className="circle_li__item_img">3</span>
                            <span className="circle_li__item_text">Получите автомобиль</span>
                        </div>
                    </div>
                </div>
            </section>
            <section>
                <div className="content-container">
                    <h2>У вас есть автомобиль?</h2>
                    <p>Чтобы он не простаивал — сдавайте его в аренду и зарабатывайте.</p>
                    <div className="circle_li__wrap">
                        <div className="circle_li__item_container">
                            <span className="circle_li__item_img"><img src={birka}/></span>
                            <span className="circle_li__item_text">Вы сами указываете цену</span>
                        </div>
                        <div className="circle_li__item_container">
                            <span className="circle_li__item_img"><img src={dollar}/></span>
                            <span className="circle_li__item_text">Мы страхуем автомобили</span>
                        </div>
                        <div className="circle_li__item_container">
                            <span className="circle_li__item_img"><img src={percent}/></span>
                            <span className="circle_li__item_text">Наша комиссия всего 3%</span>
                        </div>
                        <div className="circle_li__item_container">
                            <span className="circle_li__item_img"><img src={money}/></span>
                            <span className="circle_li__item_text">Выплаты каждую неделю</span>
                        </div>
                        {/*<div className="dashed-border-div"></div>*/}
                    </div>
                </div>
            </section>
            <section className="bg-gray">
                <div className="content-container">
                    <h2>Отзывы клиентов</h2>
                </div>
                <p style={{display: "block", textAlign: "center"}}>Пока что нет ни одного отзыва</p>
            </section>
            <section>
                <div className="content-container">
                    <img src={toy_car}/>
                    <h2>Попробуйте аренду на себе</h2>
                    <NavLink to="/register/1" className="register__main__below">
                        Зарегистрироваться
                    </NavLink>
                </div>
            </section>
            <Footer />
        </>
    )
}

export default MainUnAuth;
