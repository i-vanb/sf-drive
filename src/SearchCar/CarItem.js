import React from 'react';


export const CarItem = props => {
    const {name, price, imgAvatar, imgCar} = props;

    return (
        <div className="car-item">
            <img className="bgCar" src={imgCar} />
            <div className="carAvatarWrapper">
                <img className="carUserAvatar" src={imgAvatar}/>
            </div>
            <div className="text">
                <span className="carName"><b>{name}</b></span>
                <span className="carPrice">{"от " + price + " ₽/сутки"}</span>
            </div>
        </div>
    )
}
