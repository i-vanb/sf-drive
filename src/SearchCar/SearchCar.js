import React, {useState} from "react";
import DatePicker from "react-datepicker"
import "react-datepicker/dist/react-datepicker.css";
import calendarIcon from "../img/calendar-icon.svg"
import {getFilteredCars} from "../redux/action/cars";
import {useDispatch} from "react-redux";
import {useQuery} from "@apollo/client";
import {FETCH_CARS_BY_CITY_QUERY} from "../utils/graphql-request";
import {fetchData} from "../api";
import {getUserCars} from "../redux/action/user";
import {setCreated} from "../redux/action/car";
import axios from "axios";
import {resetLoading} from "../redux/action/system";


export const SearchCar = ({onSearch}) => {
    const [dateRange, setDateRange] = useState([null, null]);
    const [startDate, endDate] = dateRange;
    const [cityInput, setCityInput] = useState('')

    const filterHandler = e => {
        e.preventDefault()
        onSearch({city: cityInput})
        // setCityInput('')
    }

    return (
        <section>
            <div className="content-container">
                <h1>Арендуйте автомобиль</h1>
                <form className="search_car__form">
                    <div className="form_content">
                        <input placeholder="" className="form_input" value={cityInput}
                               onChange={e=>setCityInput(e.target.value)}/>
                        <span>Местоположение</span>
                    </div>
                    <div className="form_content">
                        {/*<img src={calendarIcon} className="calendarIcon"/>*/}
                        {/*<input placeholder="" className="form_input"/>*/}
                        <DatePicker
                            // className="search_car__form__datePicker"
                            selectsRange={true}
                            startDate={startDate}
                            endDate={endDate}
                            onChange={(update) => {
                                setDateRange(update);
                            }}
                        />

                        <span>Период аренды</span>
                    </div>
                    <div className="form_content">
                        <input placeholder="" className="form_input last-input-circle"/>
                        <span>Категория</span>
                    </div>
                    <div className="submit">
                        <button onClick={filterHandler} className="submit__btn">Найти</button>
                    </div>
                </form>
            </div>
            <h2 className="search_title">Рекомендуем поблизости</h2>
            {/*<TestFileInput />*/}
        </section>
    )
}


const TestFileInput = () => {
    const [fileInput, setFileInput] = useState('')
    const [photos, setPhotos] = useState([])
    console.log(photos)

    const fileUploadHandler = async () => {
        const formData = new FormData();
        formData.append(
            "file",
            fileInput, fileInput.name
        );

        const data = await axios.post('http://localhost:8000/file/create', formData)
        console.log(data)
            // .then((data)=>{
            //     console.log(data)
            // })
            // .catch(({response})=>{
            //     console.log(data)
            // })
    }

    const saveFileHandler = e => {
        const file = e.target.files[0]
        setFileInput(file)
    }

    const getAllFiles = async () => {
        const data = axios.get('http://localhost:8000/file/all')
            .then((data)=>{
              console.log(data)
                setPhotos(data.data)
            })
            .catch(({response})=>{
                console.log(data)
            })
    }

    return(
        <>
            <input type="file" onChange={saveFileHandler} />
            <button onClick={fileUploadHandler}>ЗАГРУЗИТЬ</button>
            <button onClick={getAllFiles}>Файлы</button>
            <div>{photos.map(i => <img key={i.id} src={`data:${i.mimetype};base64,${Buffer.from(i.buffer).toString('base64')}`}
                />)}</div>
        </>
    )
}


