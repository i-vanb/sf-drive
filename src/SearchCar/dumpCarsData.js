export const carList = [
    {
        name: "BMW 3-series, 2019",
        imgAvatar: "cars/img/ava1.png",
        imgCar: "cars/img/imgCar1.png",
        price: 2000,
        id: 1
    },
    {
        name: "Kia Optima, 2017",
        imgAvatar: "cars/img/ava2.png",
        imgCar: "cars/img/imgCar2.png",
        price: 2500,
        id: 2
    },
    {
        name: "Hyundai Solaris, 2018",
        imgAvatar: "cars/img/ava3.png",
        imgCar: "cars/img/imgCar3.png",
        price: 1800,
        id: 3
    },
    {
        name: "Kia Sportage, 2017",
        imgAvatar: "cars/img/ava4.png",
        imgCar: "cars/img/imgCar4.png",
        price: 3000,
        id: 4
    },
    {
        name: "Lada Granta, 2019",
        imgAvatar: "cars/img/ava5.png",
        imgCar: "cars/img/imgCar5.png",
        price: 900,
        id: 5
    },
    {
        name: "Mercedes S-classe, 2019",
        imgAvatar: "cars/img/ava6.png",
        imgCar: "cars/img/imgCar6.png",
        price: 5000,
        id: 6
    },
    {
        name: "Mazda CX-5, 2019",
        imgAvatar: "cars/img/ava7.png",
        imgCar: "cars/img/imgCar7.png",
        price: 3900,
        id: 7
    },
    {
        name: "Toyota Camry, 2015",
        imgAvatar: "cars/img/ava8.png",
        imgCar: "cars/img/imgCar8.png",
        price: 4000,
        id: 8
    },
    {
        name: "Lada Vesta, 2018",
        imgAvatar: "cars/img/ava9.png",
        imgCar: "cars/img/imgCar9.png",
        price: 129,
        id: 9
    },
    {
        name: "Nissan Almera, 2017",
        imgAvatar: "cars/img/ava10.png",
        imgCar: "cars/img/imgCar10.png",
        price: 1300,
        id: 10
    },
    {
        name: "Toyota Rav 4, 2020",
        imgAvatar: "cars/img/ava11.png",
        imgCar: "cars/img/imgCar11.png",
        price: 5000,
        id: 11
    },
    {
        name: "Kia Rio, 2020",
        imgAvatar: "cars/img/ava12.png",
        imgCar: "cars/img/imgCar12.png",
        price: 3000,
        id: 12
    },
]
