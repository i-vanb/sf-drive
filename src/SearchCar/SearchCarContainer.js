import React, {useEffect, useState} from "react";
import {withRouter} from "react-router-dom";
import {connect} from "react-redux";
import Footer from "../components/Footer";
import DatePicker from "react-datepicker"
import "react-datepicker/dist/react-datepicker.css";
import calendarIcon from "../img/calendar-icon.svg"
import {SearchCar} from "./SearchCar";
import {carList} from "./dumpCarsData";
import {CarItem} from "./CarItem";
import {CarDetail} from "./CarDetail";
import {useQuery} from "@apollo/client";
import {FETCH_All_CARS_QUERY, FETCH_CARS_BY_CITY_QUERY} from "../utils/graphql-request";
import {compose} from "redux";


const SearchCarContainer = props => {
    const [state, setState] = useState({
        city: "",
        dateBegin: "",
        dateFinish: "",
        type: "",
        startDate: null,
        endDate: null,
        detail: false,
        carItem: {}
    })


    useEffect(()=>{
    }, [])

    const filterHandler = params => {
        setState({...state, ...params})
    }

    const backFromDetailHandler = () => {
        setState({...state, carItem: {}, detail: false})
    }

    const onDetailHandler = car => {
        props.history.push(`/car/detail/${car.id}`)
    }

    return(
        <>
            <SearchCar onSearch={filterHandler}/>
            <CarsListWrapper city={state.city} onDetail={onDetailHandler} auth={props.state}/>
            <Footer/>
        </>
    )
}

const mapStateToProps = state => ({
    state: state.auth
})

export default compose(withRouter, connect(mapStateToProps, {}))(SearchCarContainer)

export const CarsListWrapper = ({city, onDetail, auth}) => {
    const {loading, data} = useQuery(FETCH_CARS_BY_CITY_QUERY, {
        variables: {city}
    })

    if(loading) return null

    return <CarMainList cars={Object.values(data)[0].filter(i => parseInt(i.ownerId) !== parseInt(auth.userID))}
                        getDetails={onDetail} />


    // return(
    //     <div className="cars">
    //         {!loading && Object.values(data)[0].map((car, index) => {
    //             if(auth.userID == car.ownerId) return null
    //             return(
    //                 <a key={index} onClick={() => onDetail(car)}><CarItem {...car} /></a>
    //             )
    //         })}
    //     </div>
    // )
}


export const CarMainList = ({cars, getDetails}) => {
    return(
        <div className="cars">
            {cars.map((car, index) => {
                return(
                    <a key={index} onClick={() => getDetails(car)}><CarItem {...car} /></a>
                )
            })}
        </div>
    )
}
