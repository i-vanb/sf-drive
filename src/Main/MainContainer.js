import React from "react";
import {connect} from "react-redux";
import MainUnAuth from "./Main";
import SearchCarContainer from "../SearchCar/SearchCarContainer";


class MainContainer extends React.Component {


    render() {
        if(this.props.state.isAuthorized) return <SearchCarContainer />
        return <MainUnAuth />
    }
}

const mapStateToProps = state => ({
    state: state.auth
})

export default connect(mapStateToProps, {})(MainContainer)
