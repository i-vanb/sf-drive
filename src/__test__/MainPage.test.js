import {CarMainList} from "../SearchCar/SearchCarContainer";
import {shallow} from "enzyme";
import Main from "../Main/Main";

const mockCars = [{}]
const mockGetDetails = jest.fn()


describe('Car List', function () {
    it('should be rendered', () => {
        const wrapper = shallow(<CarMainList cars={mockCars} getDetails={mockGetDetails}/>)
        expect(wrapper).toBeDefined()
    });
    it('Detail function should called on click', () => {
        const wrapper = shallow(<CarMainList cars={mockCars} getDetails={mockGetDetails}/>)
        wrapper.find('a').simulate('click')
        expect(mockGetDetails).toBeCalled()
    });
    it('snapshot matching', () => {
        const wrapper = shallow(<CarMainList cars={mockCars} getDetails={mockGetDetails}/>)
        expect(wrapper).toMatchSnapshot()
    });
});

describe('Unauthorized Main page', () => {
    it('snapshot matching', () => {
        const wrapper = shallow(<Main />)
        expect(wrapper).toMatchSnapshot()
    });
});
