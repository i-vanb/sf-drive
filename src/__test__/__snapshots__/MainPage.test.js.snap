// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Car List snapshot matching 1`] = `
<div
  className="cars"
>
  <a
    key="0"
    onClick={[Function]}
  >
    <CarItem />
  </a>
</div>
`;

exports[`Unauthorized Main page snapshot matching 1`] = `
<Fragment>
  <section
    className="main"
  >
    <div
      className="content-container"
    >
      <img
        alt="main picture"
        className="main_img_mob"
        src="main-car-mob.svg"
      />
      <h1>
        Каршеринг в любой
      </h1>
      <h1>
        точке России
      </h1>
      <br />
      <p
        className="main__text"
      >
        Будьте всегда за рулём во время путешествий
      </p>
      <p
        className="main__text"
      >
        и командировок.
      </p>
      <div
        className="register__main__header"
      >
        <NavLink
          to="/register/1"
        >
          Зарегистрироваться
        </NavLink>
      </div>
    </div>
  </section>
  <section>
    <div
      className="content-container row"
    >
      <div>
        <img
          alt="main picture"
          src="main_2.svg"
        />
      </div>
      <div
        className="desc"
      >
        <h2>
          Аренда напрямую от владельцев
        </h2>
        <p>
          Вы получите автомобиль от его собственника, а мы проверим юридическую чистоту и техническую исправность.
        </p>
      </div>
    </div>
  </section>
  <section>
    <div
      className="content-container row"
    >
      <div
        className="desc"
      >
        <h2>
          Автомобили на любой вкус
        </h2>
        <p>
          Вы всегда можете подобрать автомобиль любого класса от бюджетных моделей до премиум-класса и спорткаров.
        </p>
      </div>
      <div>
        <img
          alt="main picture"
          src="main_3.svg"
        />
      </div>
    </div>
  </section>
  <section>
    <div
      className="content-container row"
    >
      <div>
        <img
          alt="main picture"
          src="main_4.svg"
        />
      </div>
      <div
        className="desc"
      >
        <h2>
          Гарантия честной аренды
        </h2>
        <p>
          Общение и оплата происходит через наш сервис, что предотвращает вас от обмана.
        </p>
      </div>
    </div>
  </section>
  <section
    className="bg-gray"
  >
    <div
      className="content-container"
    >
      <h2>
        Как арендовать автомобиль
      </h2>
      <div
        className="circle_li__wrap"
      >
        <div
          className="circle_li__item_container"
        >
          <span
            className="circle_li__item_img"
          >
            1
          </span>
          <span
            className="circle_li__item_text"
          >
            Выберите автомобиль
          </span>
        </div>
        <div
          className="circle_li__item_container"
        >
          <span
            className="circle_li__item_img"
          >
            2
          </span>
          <span
            className="circle_li__item_text"
          >
            Забронируйте дату и время
          </span>
        </div>
        <div
          className="circle_li__item_container"
        >
          <span
            className="circle_li__item_img"
          >
            3
          </span>
          <span
            className="circle_li__item_text"
          >
            Получите автомобиль
          </span>
        </div>
      </div>
    </div>
  </section>
  <section>
    <div
      className="content-container"
    >
      <h2>
        У вас есть автомобиль?
      </h2>
      <p>
        Чтобы он не простаивал — сдавайте его в аренду и зарабатывайте.
      </p>
      <div
        className="circle_li__wrap"
      >
        <div
          className="circle_li__item_container"
        >
          <span
            className="circle_li__item_img"
          >
            <img
              src="birka.svg"
            />
          </span>
          <span
            className="circle_li__item_text"
          >
            Вы сами указываете цену
          </span>
        </div>
        <div
          className="circle_li__item_container"
        >
          <span
            className="circle_li__item_img"
          >
            <img
              src="dollar.svg"
            />
          </span>
          <span
            className="circle_li__item_text"
          >
            Мы страхуем автомобили
          </span>
        </div>
        <div
          className="circle_li__item_container"
        >
          <span
            className="circle_li__item_img"
          >
            <img
              src="percent.svg"
            />
          </span>
          <span
            className="circle_li__item_text"
          >
            Наша комиссия всего 3%
          </span>
        </div>
        <div
          className="circle_li__item_container"
        >
          <span
            className="circle_li__item_img"
          >
            <img
              src="money.svg"
            />
          </span>
          <span
            className="circle_li__item_text"
          >
            Выплаты каждую неделю
          </span>
        </div>
      </div>
    </div>
  </section>
  <section
    className="bg-gray"
  >
    <div
      className="content-container"
    >
      <h2>
        Отзывы клиентов
      </h2>
    </div>
    <p
      style={
        Object {
          "display": "block",
          "textAlign": "center",
        }
      }
    >
      Пока что нет ни одного отзыва
    </p>
  </section>
  <section>
    <div
      className="content-container"
    >
      <img
        src="toy_car.svg"
      />
      <h2>
        Попробуйте аренду на себе
      </h2>
      <NavLink
        className="register__main__below"
        to="/register/1"
      >
        Зарегистрироваться
      </NavLink>
    </div>
  </section>
  <Footer />
</Fragment>
`;
